drv/lpc55-sprot-server/src/handler.rs

```diff
@@ -50,213 +54,125 @@ impl Handler {
-                    .update
-                    .block_size()
-                    .map(|size| Some(size.try_into().unwrap_lite()))
-                    .map_err(|err| err.into());
-                tx_buf.serialize(MsgType::UpdBlockSizeRsp, rsp)
-            }
-            MsgType::UpdPrepImageUpdateReq => {
-                match hubpack::deserialize::<UpdateTarget>(rx_payload) {
-                    Ok((image_type, _n)) => {
-                        let rsp: UpdateRspHeader = self
-                            .update
-                            .prep_image_update(image_type)
-                            .map(|_| None)
-                            .map_err(|e| e.into());
-                        tx_buf.serialize(MsgType::UpdPrepImageUpdateRsp, rsp)
-                    }
-                    Err(e) => Err((tx_buf, e.into())),
-                }
-            }
-            MsgType::UpdWriteOneBlockReq => {
-                match hubpack::deserialize::<u32>(rx_payload) {
-                    Ok((block_num, block)) => {
-                        let rsp: UpdateRspHeader = self
-                            .update
-                            .write_one_block(block_num as usize, block)
-                            .map(|_| None)
-                            .map_err(|e| e.into());
-
-                        tx_buf.serialize(MsgType::UpdWriteOneBlockRsp, rsp)
-                    }
-                    Err(e) => Err((tx_buf, e.into())),
-                }
-            }
-
-            MsgType::UpdAbortUpdateReq => {
-                let rsp: UpdateRspHeader = self
-                    .update
-                    .abort_update()
-                    .map(|_| None)
-                    .map_err(|e| e.into());
-                tx_buf.serialize(MsgType::UpdAbortUpdateRsp, rsp)
-            }
-            MsgType::UpdFinishImageUpdateReq => {
-                let rsp: UpdateRspHeader = self
-                    .update
-                    .finish_image_update()
-                    .map(|_| None)
-                    .map_err(|e| e.into());
-                tx_buf.serialize(MsgType::UpdFinishImageUpdateRsp, rsp)
-            }
-            MsgType::SinkReq => {
-                // The first two bytes of a SinkReq payload are the U16
-                // mod 2^16 sequence number.
-                if rx_payload.len() >= 2 {
-                    let tx_payload = tx_buf.payload_mut();
-                    tx_payload[..2].copy_from_slice(&rx_payload[..2]);
-                    tx_buf.from_existing(MsgType::SinkRsp, 2)
-                } else {
-                    Ok(tx_buf.no_payload(MsgType::SinkRsp))
-                }
-            }
-            MsgType::DumpReq => {
-                let addr =
-                    u32::from_le_bytes(rx_payload[0..4].try_into().unwrap());
+            ReqBody::IoStats => Ok(RspBody::IoStats(stats.clone())),
+            ReqBody::Sprockets(req) => {
+                // TODO: Don't unwrap!
```

Comment:
Since you have context, would you mind adding a sentence to this comment explaining how you'd avoid unwrapping here? What's the potential panic?

Reply:
Fixed by not unwrapping at all and leaving a comment.